core/arch/arm/kernel/timer_a64.c

```diff
@@ -60,7 +60,7 @@ void generic_timer_handler(uint32_t time_ms)
 	write_cntps_ctl(0);
 
 	/* Reconfigure timer to fire time_ms from now */
-	timer_ticks = (read_cntfrq() * time_ms) / 1000;
+	timer_ticks = ((uint64_t)read_cntfrq() * time_ms) / 1000;
```

Comment:
Equivalent fix is needed in `generic_timer_start()`.

By the way, could you also  initialize local `timer_ticks` variables in both function, as a cleanup to satisfy the coding style rules?

Reply:
Good point, I'll fix it.